File: app/src/main/java/com/qiscus/sdksample/qiscussamplesdk/NewChatDialog.java
package com.qiscus.sdksample.qiscussamplesdk;

import android.app.Dialog;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.support.v4.util.PatternsCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;
import android.widget.TextView;

public class NewChatDialog extends DialogFragment implements View.OnClickListener {

    private TextView nameField;
    private TextView emailField;
    private View submitButton;
    private View cancelButton;

    private Listener listener;

    public static NewChatDialog newInstance(Listener listener) {
        NewChatDialog dialog = new NewChatDialog();
        dialog.listener = listener;
        dialog.setCancelable(false);
        return dialog;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = LayoutInflater.from(getActivity()).inflate(R.layout.dialog_new_chat, container, false);
        nameField = (TextView) view.findViewById(R.id.name);
        emailField = (TextView) view.findViewById(R.id.email);
        submitButton = view.findViewById(R.id.tv_submit);
        cancelButton = view.findViewById(R.id.tv_cancel);

        submitButton.setOnClickListener(this);
        cancelButton.setOnClickListener(this);

        return view;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Dialog dialog = super.onCreateDialog(savedInstanceState);
        dialog.getWindow().requestFeature(Window.FEATURE_NO_TITLE);
        return dialog;
    }

    @Override
    public void onStart() {
        super.onStart();
        Dialog dialog = getDialog();
        if (dialog != null) {
            dialog.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
            dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
            dialog.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN);
        }
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.tv_submit:
                String name = nameField.getText().toString();
                String email = emailField.getText().toString();
                if (name.isEmpty()) {
                    nameField.setError("Please insert name!");
                    nameField.requestFocus();
                } else if (!PatternsCompat.EMAIL_ADDRESS.matcher(email).matches()) {
                    emailField.setError("Please insert a valid email!");
                    emailField.requestFocus();
                } else {
                    if (listener != null) {
                        listener.onSubmit(name, email);
                    }
                    dismiss();
                }
                break;
            case R.id.tv_cancel:
                dismiss();
                break;
        }
    }


    public interface Listener {
        void onSubmit(String name, String email);
    }
}

File: app/src/main/java/com/qiscus/sdksample/qiscussamplesdk/NewChatDialog.kt
package com.qiscus.sdksample.qiscussamplesdk

import android.app.Dialog
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v4.util.PatternsCompat
import android.view.*
import android.widget.TextView

class NewChatDialog : DialogFragment(), View.OnClickListener {

    private var nameField: TextView? = null
    private var emailField: TextView? = null
    private var submitButton: View? = null
    private var cancelButton: View? = null

    private var listener: Listener? = null

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = LayoutInflater.from(activity).inflate(R.layout.dialog_new_chat, container, false)
        nameField = view.findViewById(R.id.name) as TextView
        emailField = view.findViewById(R.id.email) as TextView
        submitButton = view.findViewById(R.id.tv_submit)
        cancelButton = view.findViewById(R.id.tv_cancel)

        submitButton!!.setOnClickListener(this)
        cancelButton!!.setOnClickListener(this)

        return view
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dialog = super.onCreateDialog(savedInstanceState)
        dialog.window!!.requestFeature(Window.FEATURE_NO_TITLE)
        return dialog
    }

    override fun onStart() {
        super.onStart()
        val dialog = dialog
        if (dialog != null) {
            dialog.window!!.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
            dialog.window!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
            dialog.window!!.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN)
        }
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.tv_submit -> {
                val name = nameField!!.text.toString()
                val email = emailField!!.text.toString()
                if (name.isEmpty()) {
                    nameField!!.error = "Please insert name!"
                    nameField!!.requestFocus()
                } else if (!PatternsCompat.EMAIL_ADDRESS.matcher(email).matches()) {
                    emailField!!.error = "Please insert a valid email!"
                    emailField!!.requestFocus()
                } else {
                    if (listener != null) {
                        listener!!.onSubmit(name, email)
                    }
                    dismiss()
                }
            }
            R.id.tv_cancel -> dismiss()
        }
    }


    interface Listener {
        fun onSubmit(name: String, email: String)
    }

    companion object {

        fun newInstance(listener: Listener): NewChatDialog {
            val dialog = NewChatDialog()
            dialog.listener = listener
            dialog.isCancelable = false
            return dialog
        }
    }
}
